app/routers/dogfinder.py

```diff
@@ -108,6 +113,11 @@ class QueryRequest(BaseModel):
     breed: Optional[str] = None
     top: int = 10
     return_properties: Optional[List[str]] = ["type", "breed", "filename", "image", IS_FOUND_FIELD, DOG_ID_FIELD, "contactName", "contactPhone", "contactEmail", "contactAddress"]
+    
+    def __init__(self, **data):
+        # Make sure the query holds the correct type before querying the VDB
+        data["type"] = DogType.LOST.value if data["type"] == DogType.FOUND.value else DogType.FOUND.value
```

Comment:
Is it possible to change the type class in the query and add document from str to this enum to better support the logic?
Also can you log the data['type'] received so we can debug it in case of issues since this logic has protection against stuff like casing or typos

Reply:
Yes. If I understand you correctly, you want to also validate this type in the api. 
AFAIK FastApi supports pydantic and enums really well. How about we change it in here:
![Screen Shot 2023-10-16 at 9 12 34](https://github.com/NirDiamant/dog_finder/assets/80447963/f3800a45-f200-430d-8851-8c1bb7079cf1)

to `type: DogType = Form(...)` and we can also do this in the `add_document`? 
Is this what you meant in your comment to begin with?

And yeah, I'll log the data['type'] before I change it.

Reply:
Changed it. Please review the new changes. This looks like this in swagger:

![Uploading Screen Shot 2023-10-16 at 9.39
![Screen Shot 2023-10-16 at 9 41 07](https://github.com/NirDiamant/dog_finder/assets/80447963/3445fdfe-7e1b-45f2-beef-11297defb774)
.31.png…]()

Don't know if this is what you meant, but I think it's best that is also works as validation layer